src/data/automation_i18n.ts

```diff
@@ -141,23 +145,62 @@ export const describeTrigger = (
 
     base += ` ${entities} changes`;
 
-    if (trigger.from) {
+    const stateObj =
+      hass.states[
+        Array.isArray(trigger.entity_id)
+          ? trigger.entity_id[0]
+          : trigger.entity_id
+      ];
+    if (trigger.from !== undefined) {
```

Comment:
Do you need to exclude the `trigger.from === null` case? Otherwise I think it will throw an exception below for trigger.from.toString(). (I'm touching similar in #15071)

Reply:
Yes we do need it as it has meaning. But nice catch, will address that!

Reply:
Adjusted accordingly. Added screenshots to the OP.